torch_xla/experimental/custom_kernel.py

```diff
@@ -331,6 +331,51 @@ def flash_attention(
   return FlashAttention.apply(q, k, v, causal)
 
 
+def paged_attention(q, k_pages, v_pages, lengths, page_indices,
```

Comment:
The original kernel has this thing called: q_dtype_for_kernel_launch? What does it do? Should we copy that as well?

Reply:
In the original kernel, the `q_dtype_for_kernel_launch` is always either `jnp.float32` or q's `dtype`. In our case, I'm expecting the passed-in q's `dtype` to be `torch.32`, so the `q_dtype_for_kernel_launch` will always be `float32`.

Reply:
No, I don't think that will be the case for actual workflow. It could be bf16 or even in8 etc...

Reply:
I see, makes sense. Just updated to handle `q_dtype_for_kernel_launch`, following jax's kernel -- https://github.com/google/jax/blob/main/jax/experimental/pallas/ops/tpu/paged_attention/paged_attention_kernel.py#L393. I can follow-up in another PR to add some more unit tests for different dtypes for `q`.